test/AuthBridgeExecutor.t.sol

```diff
@@ -0,0 +1,543 @@
+        skip(DELAY);
+
+        vm.expectRevert(abi.encodeWithSignature("NotGuardian()"));
+        executor.cancel(0);
+    }
+
+    function test_cancel_actionsSetIdTooHigh_boundary() public {
+        assertEq(executor.getActionsSetCount(), 0);
+        vm.expectRevert(abi.encodeWithSignature("InvalidActionsSetId()"));
+        vm.prank(guardian);
+        executor.cancel(0);
+
+        _queueAction();
+        skip(DELAY);
+
+        assertEq(executor.getActionsSetCount(), 1);
+        vm.prank(guardian);
+        executor.cancel(0);
+    }
+
+    function test_cancel_notQueued_cancelled() public {
+        _queueAction();
+        vm.prank(guardian);
+        executor.cancel(0);
+        
+        vm.expectRevert(abi.encodeWithSignature("OnlyQueuedActions()"));
+        vm.prank(guardian);
+        executor.cancel(0);
+    }
+
+    function test_cancel_notQueued_executed() public {
+        _queueAction();
+        skip(DELAY);
+
+        vm.prank(guardian);
+        executor.cancel(0);
+        
+        vm.expectRevert(abi.encodeWithSignature("OnlyQueuedActions()"));
+        vm.prank(guardian);
+        executor.cancel(0);
+    }
+
+    function test_cancel_notQueued_expired_boundary() public {
+        _queueAction();
+        skip(DELAY + GRACE_PERIOD + 1);
+        
+        vm.expectRevert(abi.encodeWithSignature("OnlyQueuedActions()"));
+        vm.prank(guardian);
+        executor.cancel(0);
+
+        vm.warp(block.timestamp - 1);
+
+        vm.prank(guardian);
+        executor.cancel(0);
+    }
+
+    function test_cancel() public {
+        Action memory action = _getDefaultAction();
+        bytes32 actionHash = _encodeHash(action, block.timestamp + DELAY);
+        _queueAction(action);
+
+        assertEq(executor.isActionQueued(actionHash),    true);
+        assertEq(executor.getActionsSetById(0).canceled, false);
+        assertEq(uint8(executor.getCurrentState(0)),     uint8(IExecutorBase.ActionsSetState.Queued));
+
+        vm.expectEmit(address(executor));
+        emit ActionsSetCanceled(0);
+        vm.prank(guardian);
+        executor.cancel(0);
+
+        assertEq(executor.isActionQueued(actionHash),    false);
+        assertEq(executor.getActionsSetById(0).canceled, true);
+        assertEq(uint8(executor.getCurrentState(0)),     uint8(IExecutorBase.ActionsSetState.Canceled));
+    }
+
+    function test_updateGuardian_notSelf() public {
+        vm.expectRevert(abi.encodeWithSignature("OnlyCallableByThis()"));
+        executor.updateGuardian(guardian);
+    }
+
+    function test_updateGuardian() public {
+        address newGuardian = makeAddr("newGuardian");
+
+        assertEq(executor.getGuardian(), guardian);
+
+        vm.expectEmit(address(executor));
+        emit GuardianUpdate(guardian, newGuardian);
+        vm.prank(address(executor));
+        executor.updateGuardian(newGuardian);
+
+        assertEq(executor.getGuardian(), newGuardian);
+    }
+
+    function test_updateDelay_notSelf() public {
```

Comment:
Need failure mode coverage for this

Reply:
Will be removing the validation for min/max in subsequent PR, so I intentionally skipped it.